pqarrow/arrowutils/sort.go

```diff
@@ -0,0 +1,108 @@
+package arrowutils
+
+import (
+	"bytes"
+	"fmt"
+	"sort"
+
+	"github.com/apache/arrow/go/v14/arrow"
+	"github.com/apache/arrow/go/v14/arrow/array"
+	"github.com/apache/arrow/go/v14/arrow/memory"
+	"golang.org/x/exp/constraints"
+
+	"github.com/polarsignals/frostdb/pqarrow/builder"
+)
+
+// SortRecord sorts the given record's rows by the given column. Currently only supports int64, string and binary columns.
+func SortRecord(r arrow.Record, cols []int) ([]int, error) {
+	if len(cols) > 1 {
+		return nil, fmt.Errorf("sorting by multiple columns isn't implemented yet")
+	}
+	if r.NumRows() == 0 {
+		return nil, nil
+	}
+	if r.NumRows() == 1 {
+		return []int{0}, nil
+	}
+
+	indices := make([]int, r.NumRows())
+	// populate indices
+	for i := range indices {
+		indices[i] = i
+	}
+
+	switch c := r.Column(cols[0]).(type) {
+	case *array.Int64:
+		sort.Sort(orderedSorter[int64]{array: c, indices: indices})
+	case *array.String:
+		sort.Sort(orderedSorter[string]{array: c, indices: indices})
+	case *array.Binary:
+		sort.Sort(binarySort{array: c, indices: indices})
+	default:
+		return nil, fmt.Errorf("unsupported column type for sorting %T", c)
+	}
+
+	return indices, nil
+}
+
+// ReorderRecord reorders the given record's rows by the given indices.
+func ReorderRecord(mem memory.Allocator, r arrow.Record, indices []int) (arrow.Record, error) {
+	// if the indices are already sorted, we can return the original record to save memory allocations
+	if sort.SliceIsSorted(indices, func(i, j int) bool { return indices[i] < indices[j] }) {
+		return r, nil
+	}
+
+	recordBuilder := builder.NewRecordBuilder(mem, r.Schema())
```

Comment:
`compute.Take` supports `arrow.Record` I was wondering why you chose not to use it.

Reply:
I wasn't aware of it and this is an old version from months ago. We should take the time to refactor it!